lib/easy.c

```diff
@@ -84,6 +84,27 @@
 #include "curl_memory.h"
 #include "memdebug.h"
 
+
+/* Curl_use_data_blob is used by curl library to
+ *  check if a curl parameter is a memory blob instead filename
+ * it return 1 and fill len and data if this is memory blob
+ * and return 0 if this is a filename (or another string) */
+bool Curl_use_data_blob(const struct curl_blob* str, size_t *len, void **data)
+{
+  if(!str)
+    return false;
+  struct curl_blob* st_blob = (struct curl_blob*)str;
```

Comment:
Is this line necessary? Why can't you access str directly?

Reply:
I just fix it  This was heritage of previous version :-)

Reply:
I've spotted similar things like this in the code as well. I would suggest you to search in the file for `(struct curl_blob*)`